### PR TITLE
[SL-148] Layout listing updates

### PR DESCRIPTION
**Pull request by rafsuntaskin**
### 🎫 Ticket

[SL-148]

### 🗒️ Description
* Layout `Edit` link opens up iframe view to edit the layout.
* Changed the layout card text “No upcoming events” to “No associated events” when a layout has no associated events of any kind.
* Count and display the number of associated events when available and hide the `delete` button.

### 🎥 Artifacts 
📷 screenshot(s): https://share.cleanshot.com/10qqbFBK

### ✔️ Checklist
- [ ] Changelog entry in the `readme.txt` file.
- [ ] Code is covered by **NEW** `wpunit` or `integration` tests.
- [ ] Code is covered by **EXISTING** `wpunit` or `integration` tests.
- [ ] Are all the **required** tests passing?
- [ ] Automated code review comments are addressed.
- [ ] Have you added Artifacts?
- [ ] Check the base branch for your PR.
- [ ] Add your PR to the project board for the release.

[SL-148]: https://stellarwp.atlassian.net/browse/SL-148?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: tests/slr_integration/Admin/__snapshots__/Maps_Layout_Homepage_Test__test_layouts_tab_card_listing__0.snapshot.html [hunk 1]
```diff
@@ -39,14 +39,14 @@
 	</div>
 	<div class="tec-tickets__seating-tab__card-content">
 		<div class="tec-tickets__seating-tab__card-title">Layout 1</div>
-		<div class="tec-tickets__seating-tab__card-capacity">
-			100 seats		</div>
+		<div class="tec-tickets__seating-tab__card-info">
+			No associated events		</div>
```
Comment (lucatume) [inline tests/slr_integration/Admin/__snapshots__/Maps_Layout_Homepage_Test__test_layouts_tab_card_listing__0.snapshot.html:43]: Here and elsewhere in the tests, the tets case should be setting things up to have Events associated with the map, but the map is showing as not associated to anything.

File: src/Tickets/Seating/Service/Layouts.php [hunk 2]
```diff
@@ -206,4 +208,26 @@ public static function update_transient_name(): string {
...
+	 * @since TBD
+	 *
+	 * @param string $layout_id The ID of the layout.
+	 *
+	 * @return int The number of events associated with the layout.
+	 */
+	public static function get_associated_events_by_id( string $layout_id ): int {
+		try {
+			$count = DB::table( 'posts', 'posts' )
+						->leftJoin( 'postmeta', 'posts.id', 'layout_meta.post_id', 'layout_meta' )
```
Comment (lucatume) [inline src/Tickets/Seating/Service/Layouts.php:224]: Should this the be the number of Events or Occurrences? 
Should this behaviour differ depending on CT1 being active or not?

File: src/Tickets/Seating/Service/Layouts.php [hunk 3]
```diff
@@ -206,4 +208,26 @@ public static function update_transient_name(): string {
...
+	/**
+	 * Returns the number of events associated with the layout.
+	 *
+	 * @since TBD
+	 *
+	 * @param string $layout_id The ID of the layout.
+	 *
+	 * @return int The number of events associated with the layout.
+	 */
+	public static function get_associated_events_by_id( string $layout_id ): int {
```
Comment (lucatume) [inline src/Tickets/Seating/Service/Layouts.php:221]: It's "posts", not "events". 
Here and elsewhere change method names, vars and  comments to reflect the fact we're dealing with ticketable posts, that might not be Events.

**Review by lucatume — CHANGES_REQUESTED** [review]

File: tests/slr_integration/Admin/__snapshots__/Maps_Layout_Homepage_Test__test_layouts_tab_card_listing__0.snapshot.html (hunk 1 above)
Comment (rafsuntaskin) [reply tests/slr_integration/Admin/__snapshots__/Maps_Layout_Homepage_Test__test_layouts_tab_card_listing__0.snapshot.html:43]: The Maps card doesn't have any associated events [showing](https://www.figma.com/design/5TiznOUTaiwQQ5FZAgwRDm/Seat-Charts?node-id=2503-180691).

We only display associated events for [Layout Cards.](https://www.figma.com/design/5TiznOUTaiwQQ5FZAgwRDm/Seat-Charts?node-id=4243-299350)

The highlighted text is from the layouts tab -> Layout card. The maps view is not changed with this PR.

File: src/Tickets/Seating/Service/Layouts.php (hunk 2 above)
Comment (rafsuntaskin) [reply src/Tickets/Seating/Service/Layouts.php:224]: We don't have [support](https://lw.slack.com/archives/C05QHBFKYCA/p1718826994802359) for ASC tickets with recurring for now, so this should be good for the MVP.

Comment (rafsuntaskin) [reply src/Tickets/Seating/Service/Layouts.php:224]: If occurrences of an recurring event is holding the same meta data as the parent event, then this query should take care of the occurrences as well. right?

**Review by lucatume — APPROVED** [review]

**Pull request merged**